wg-cloud-provider/cloud-provider-requirements.md

```diff
@@ -0,0 +1,67 @@
+# Conventions for Cloud Provider Repositories
+
+This purpose of this document is to define a common structure for the cloud
+provider repositories owned by current and future cloud provider SIGs[1]. In
+accordance with the WG-Cloud-Provider Charter[2] to "define a set of common
+expected behaviors across cloud providers", this proposal defines the location
+and structure of commonly expected code.
```

Comment:
I'm not sure if we should add a sentence stating that this document only applies to cloud providers that have been approved by CNCF to be a part of the Kubernetes org. @calebamiles thoughts?

Reply:
Where is that list @wlan0 ? What would a cloud provider have to do to get into that list?

Reply:
I think it may be better to not call out any kind of official/approved status. I could see potentially causing issues with locating cloud providers and confusion for end users as to where the "official" version resides. Until a provider is accepted, it could cause duplication of multiple people working on the same thing. Better to have one common location where interested parties can all come together to collaborate, IMO.

Reply:
@dims https://github.com/kubernetes/community/pull/1712/files#diff-1d931bbfa82e30aa84cde953b196af78R305 

@hogepodge We'll need to communicate with @calebamiles so that there is no overlap of work. The PR above overlaps with some of the information in this PR.

> I could see potentially causing issues with locating cloud providers and confusion for end users as to where the "official" version resides.

@stmcginnis There is a well defined process to ensure that this does not happen. Please refer to the link pasted.

Reply:
@wlan0 

> I'm not sure if we should add a sentence stating that this document only applies to cloud providers that have been approved by CNCF to be a part of the Kubernetes org.

Not sure if I understood how CNCF can be influential here.

Reply:
Hi @wlan0  ,   I had the same concern with @dims , The Alibaba Cloud also want to help with the out of tree cloud provider.  And what we should do to go into that list? Is there a standard procedure?  Thanks.